### PR TITLE
Fix spotify controller and update example

### PR DESCRIPTION
**Pull request by shocklateboy92**
It seems the `setCredentials` message has changed to `addUser` at some point in the past, with a different payload.

We also discovered a new issue with playback (on our google homes and chromecasts at least) where launching the spotify app and adding the user/token to it doesn't cause it to appear in spotify's device list. This updates the example to make the same request the web player does to enable playback on the cast device.

### CODE REVIEW

**Comment by emontnemery** [comment]
@fondberg could you take a look at this please?

**Comment by emontnemery** [comment]
@shocklateboy92 , @CraigDropTables how has this been tested? The pychromecast spotify functionality is used in some other places, and since we change the required arguments for starting the app, those need to be updated too?

File: examples/spotify_example.py [hunk 1]
```diff
@@ -96,25 +105,30 @@
...
-devices_available = client.devices()
-
-# Match active spotify devices with the spotify controller's device id
-for device in devices_available["devices"]:
-    if device["id"] == sp.device:
-        spotify_device_id = device["id"]
-        break
+# The chromecast device does not show up as part of the public API get devices
+# call until it starts playing. The only way to do so is to transfer playback
+# by call this endpoint that's not part of their public API.
```
Comment (fondberg) [inline examples/spotify_example.py:110]: Hmmm. I think this is not correct.
It will should show up. I dont need any special http calls

File: pychromecast/controllers/spotify.py [hunk 2]
```diff
@@ -11,10 +11,9 @@
 APP_NAMESPACE = "urn:x-cast:com.spotify.chromecast.secure.v1"
 TYPE_GET_INFO = "getInfo"
 TYPE_GET_INFO_RESPONSE = "getInfoResponse"
-TYPE_SET_CREDENTIALS = "setCredentials"
```
Comment (fondberg) [inline pychromecast/controllers/spotify.py:14]: Why is this removed?

File: pychromecast/controllers/spotify.py [hunk 3]
```diff
@@ -11,10 +11,9 @@
...
-TYPE_SET_CREDENTIALS_ERROR = "setCredentialsError"
-TYPE_SET_CREDENTIALS_RESPONSE = "setCredentialsResponse"
-
+TYPE_ADD_USER = "addUser"
```
Comment (fondberg) [inline pychromecast/controllers/spotify.py:14]: Can you describe what this new api call does?

**Comment by shocklateboy92** [comment]
Sorry for the late response.

We tested this on our Google Home (and Google Home Mini). We originally tried it with Home Assistant, and noticed it made the "ding" noise but then got stuck. Looking at the logs, it looked like our Google Home as returning `setCredentialsError`.

So we followed the instructions on this project's readme.md and used net-export to look at what the spotify web player sends. 
That's when we noticed it doesn't send `setCredentials` at all, it sends an `addUser` in its place.

File: pychromecast/controllers/spotify.py (hunk 2 above)
Comment (shocklateboy92) [reply pychromecast/controllers/spotify.py:14]: Based on the net-export log, our web player wasn't sending this anymore. We tried it with the same auth token, and it still returned `setCredentialsError`. Doing the same `addUser` the web player was sending worked.

File: pychromecast/controllers/spotify.py (hunk 3 above)
Comment (shocklateboy92) [reply pychromecast/controllers/spotify.py:14]: As far as we can tell it does the same thing that `setCredentials` used to do, with the payload in a slightly different format.

File: examples/spotify_example.py (hunk 1 above)
Comment (shocklateboy92) [reply examples/spotify_example.py:110]: Odd. How long did it take to show up? Was it a Chromecast V1/V2? or Google Home/Mini? 
Did you use the auth token fetched by feeding your cookies to `spotify_token`?

**Comment by fondberg** [comment]
interesting. so will this fix the expiry issue after 60 mins?

File: examples/spotify_example.py (hunk 1 above)
Comment (fondberg) [reply examples/spotify_example.py:110]: I maintain spotcast so I think I represent most of the users with a variety of cast devies

File: examples/spotify_example.py [hunk 4]
```diff
@@ -96,25 +105,30 @@
...
+headers = {
+    "Accept": "application/json",
+    "Content-Type": "application/json",
+    "Authorization": "Bearer " + access_token,
+}
+transferResponse = requests.post(
+    "https://guc-spclient.spotify.com/connect-state/v1/connect/transfer/from/noop/to/{}".format(
```
Comment (fondberg) [inline examples/spotify_example.py:117]: guc is for US.

It is not advisable to use spotify internal API:s. A simple force_playback with spotipy should suffice

**Comment by rako77** [comment]
Clearly I did not test the existing functionality enough. I just tested Spotcast and it works for us so we may just need to update the example for now.

As Spotify is now using addUser instead of set credentials the setCredentials flow may break in the near future but I don't see a need to switch from setCredentials to addUser until the workflow breaks.

**Comment by fondberg** [comment]
Seperate from this but... what _would_ be interesting is to see what happens after 60 mins when sniffing. Is there a new token sent?

**Comment by emontnemery** [comment]
What's the way forwards for this PR?

**Comment by fondberg** [comment]
@emontnemery seems they were right. The old setCredentials is not working anymore. 
@rako77 can you merge master and take this forward?

**Comment by shocklateboy92** [comment]
Unfortunately @rako77 and I are a little squeezed for time at the moment.
We'll see what we can do when things calm down in a few weeks.

**Comment by fondberg** [comment]
The Spotify controller is moved to spotcast so this can be closed

**Comment by shocklateboy92** [comment]
Thanks!
